dlt/common/destination/reference.py

```diff
@@ -577,14 +579,19 @@ def with_staging_dataset(self) -> ContextManager["JobClientBase"]:
 class SupportsStagingDestination:
     """Adds capability to support a staging destination for the load"""
 
+    def __init__(self, config: DestinationClientDwhWithStagingConfiguration) -> None:
+        self.truncate_tables_on_staging_destination_before_load = (
+            config.truncate_tables_on_staging_destination_before_load
+        )
+
     def should_load_data_to_staging_dataset_on_staging_destination(
         self, table: TTableSchema
     ) -> bool:
         return False
 
     def should_truncate_table_before_load_on_staging_destination(self, table: TTableSchema) -> bool:
         # the default is to truncate the tables on the staging destination...
-        return True
+        return self.truncate_tables_on_staging_destination_before_load
```

Comment:
make this method abstract to force everyone to implement it since we can't use this nice default anymore